Sources/View/UICollectionView+Reusable.swift

```diff
@@ -37,24 +63,70 @@ public extension UICollectionView {
+   
+   - note: The `cellType` parameter can generally be omitted and infered by the return type,
+   except when your type is in a variable and cannot be determined at compile time.
+   - seealso: `dequeueReusableCell(withReuseIdentifier:,for:)`
+   */
+  final func dequeueReusableCell<T: UICollectionViewCell>(for indexPath: IndexPath, cellType: T.Type = T.self) -> T
+    where T: Reusable & AutoRegistering {
+      if !hasRegistered(cellType: cellType) {
+        register(cellType: cellType)
+      }
+      return dequeueReusableCellInternal(for: indexPath)
+  }
 
+  /**
+   Returns a reusable `UICollectionViewCell` object for the class inferred by the return-type
+   
+   - parameter indexPath: The index path specifying the location of the cell.
+   - parameter cellType: The cell class to dequeue
+   
    - returns: A `Reusable`, `UICollectionViewCell` instance
+   
+   - note: The `cellType` parameter can generally be omitted and infered by the return type,
+   except when your type is in a variable and cannot be determined at compile time.
+   - seealso: `dequeueReusableCell(withReuseIdentifier:,for:)`
+   */
+  final func dequeueReusableCell<T: UICollectionViewCell>(for indexPath: IndexPath, cellType: T.Type = T.self) -> T
+    where T: NibReusable & AutoRegistering {
+      if !hasRegistered(cellType: cellType) {
+        register(cellType: cellType)
+      }
+      return dequeueReusableCellInternal(for: indexPath)
+  }
 
+  /**
+   Returns a reusable `UICollectionViewCell` object for the class inferred by the return-type
+   
+   - parameter indexPath: The index path specifying the location of the cell.
+   - parameter cellType: The cell class to dequeue
+   
+   - returns: A `Reusable`, `UICollectionViewCell` instance
+   
    - note: The `cellType` parameter can generally be omitted and infered by the return type,
    except when your type is in a variable and cannot be determined at compile time.
    - seealso: `dequeueReusableCell(withReuseIdentifier:,for:)`
    */
   final func dequeueReusableCell<T: UICollectionViewCell>(for indexPath: IndexPath, cellType: T.Type = T.self) -> T
+    where T: Reusable {
+      return dequeueReusableCellInternal(for: indexPath)
+  }
+
+  final private func dequeueReusableCellInternal<T: UICollectionViewCell>(for indexPath: IndexPath,
+                                                                          cellType: T.Type = T.self) -> T
     where T: Reusable {
       let bareCell = self.dequeueReusableCell(withReuseIdentifier: cellType.reuseIdentifier, for: indexPath)
       guard let cell = bareCell as? T else {
         fatalError(
           "Failed to dequeue a cell with identifier \(cellType.reuseIdentifier) matching type \(cellType.self). "
             + "Check that the reuseIdentifier is set properly in your XIB/Storyboard "
-            + "and that you registered the cell beforehand"
+            + "and that you registered the cell beforehand OR enabled autoRegister"
```

Comment:
(There might be other places with that old mention left maybe, didn't read all the doc comments)

Change to ➡️ "or conformed to `AutoRegistering`"